client/client.go

```diff
@@ -56,6 +56,18 @@ import (
 	"github.com/pkg/errors"
 )
 
+// DummyHost is a hostname used for local communication.
+//
+// For local communications (npipe://, unix://), the hostname is not used,
+// but we need valid and meaningful hostname.
```

Comment:
> If the authority component is missing or undefined for the target URI, then a client MUST send a Host header field with an empty field-value.

[RFC 7230, Section 5.4](https://datatracker.ietf.org/doc/html/rfc7230#section-5.4:~:text=If%20the%20authority%20component%20is%20missing%20or%0A%20%20%20undefined%20for%20the%20target%20URI%2C%20then%20a%20client%20MUST%20send%20a%20Host%20header%0A%20%20%20field%20with%20an%20empty%20field%2Dvalue.)

The authority component is undefined for the `unix` scheme, therefore the empty string is the valid and meaningful host name to use.